pkg/asset/installconfig/powervs/session.go

```diff
@@ -90,28 +102,161 @@ func getPISession() (*ibmpisession.IBMPISession, *UserCredentials, error) {
-	// @TODO: pass through debug?
-	s, err := ibmpisession.New(passwd, region, false, defSessionTimeout, id, zone)
-	uc := &UserCredentials{UserID: id, APIKey: apikey}
-	return s, uc, err
+	// Prompt the user if a region was not found in the current enviornment context
+	if region == "" {
+		regions := knownRegions()
+
+		longRegions := make([]string, 0, len(regions))
+		shortRegions := make([]string, 0, len(regions))
+		for id, location := range regions {
+			longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
+			shortRegions = append(shortRegions, id)
+		}
+
+		sort.Strings(longRegions)
+		sort.Strings(shortRegions)
+
+		var regionTransform survey.Transformer = func(ans interface{}) interface{} {
+			switch v := ans.(type) {
+			case core.OptionAnswer:
+				return core.OptionAnswer{Value: strings.SplitN(v.Value, " ", 2)[0], Index: v.Index}
+			case string:
+				return strings.SplitN(v, " ", 2)[0]
+			}
+			return ""
+		}
+
+		err = survey.Ask([]*survey.Question{
+			{
+				Prompt: &survey.Select{
+					Message: "Region",
+					Help:    "The Power VS region to be used for installation.",
+					// Default: fmt.Sprintf("%s (%s)", defaultRegion, regions[defaultRegion]),
+					Options: longRegions,
+				},
+				Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+					choice := regionTransform(ans).(core.OptionAnswer).Value
+					i := sort.SearchStrings(shortRegions, choice)
+					if i == len(shortRegions) || shortRegions[i] != choice {
+						return errors.Errorf("invalid region %q", choice)
+					}
+					return nil
+				}),
+				Transform: regionTransform,
+			},
+		}, &region)
+		if err != nil {
+			return "", "", err
+		}
+	}
+
+	// -------------------------
+	// Zone
+	// -------------------------
+	// There is an effort to remove zone, I am uncertain how this can be done as it is required by
+	// ibmpisession, if zone is not passed we may want to assign zone to region, however I am using
+	// the code from platform.go to prompt the user for zone here instead
+
+	sessionZone = os.Getenv("IBMCLOUD_ZONE")
+
+	if len(sessionZone) > 0 {
+		if IsKnownZone(region, sessionZone) {
+			zone = sessionZone
+		} else {
+			logrus.Warnf("Unrecognized Power VS zone %s, ignoring IBMCLOUD_ZONE", sessionZone)
+		}
+	}
+
+	// Prompt the user if zone was not found in the current enviornment context
+	if zone == "" {
+		zones := knownZones(region)
+		defaultZone := zones[0]
+
+		var zoneTransform survey.Transformer = func(ans interface{}) interface{} {
+			switch v := ans.(type) {
+			case core.OptionAnswer:
+				return core.OptionAnswer{Value: strings.SplitN(v.Value, " ", 2)[0], Index: v.Index}
+			case string:
+				return strings.SplitN(v, " ", 2)[0]
+			}
+			return ""
+		}
+
+		err = survey.Ask([]*survey.Question{
+			{
+				Prompt: &survey.Select{
+					Message: "Zone",
+					Help:    "The powervs zone within the region to be used for installation.",
```

Comment:
s/powervs/Power VS.
Also -- do all the survey questions help text have punctuation? This isn't a complete sentence so I wouldn't expect a period at the end.

Reply:
I never did well in English I'm afraid.  I just wrote this out as if I was writing a description of a variable for function documentation,  Usually something like "var: The thing the var does or is"  Should I just complete the sentence?  "This is the"  Or would you prefer I just drop the period?

Reply:
I was asking about the other providers. I've seen projects not like punctuation, but I did a quick grep to see and we should always add them, it looks like. (grep -r "Help:" pkg/asset/installconfig/*)